includes/reader-revenue/woocommerce/class-woocommerce-logs.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Log WooCommerce events
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for logging WooCommerce events.
+ */
+class WooCommerce_Logs {
+	/**
+	 * Initialize the class.
+	 */
+	public static function init() {
+		add_filter( 'woocommerce_add_error', [ __CLASS__, 'log_error_notices' ] );
+	}
+
+	/**
+	 * Log error notices.
+	 *
+	 * @param string $message The error message.
+	 */
+	public static function log_error_notices( $message ) {
+		$data = [
+			'wc_cart' => WC()->cart->get_cart_contents(),
```

Comment:
Is the cart info always relevant? Should we only send this data if the error happened during a checkout?

Reply:
It's additional session context that I don't think is harmful to include. Whether it has valuable information for a particular error is up to the person debugging.

I don't see value in adding additional code to not send it.

Reply:
I think the danger in my mind is that it could cause confusion for the person debugging. This info will be included as long as anything has been added to the cart at any point during the session and then not cleared out, so the error could have nothing to do with the cart/checkout and may send the person debugging down a rabbit hole. But maybe it's something we'll want to monitor for a bit to see if that happens, rather than preemptively solving for it.